Userland/Libraries/LibSemVer/Semver.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2023, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Function.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+
+namespace SemVer {
+/**
+ * @enum BumpType
+ * Enumerates the possible types of version bumps.
+ */
+enum class BumpType {
+    Major, /**< Major version bump */
+    Minor, /**< Minor version bump */
+    Patch  /**< Patch version bump */
+};
+
+/**
+ * @enum BumpOpts
+ * Enumerates the options for version bumping.
+ */
+enum class BumpOpts {
+    KeepSuffix,      /**< Keep both -.... and +.... information during bump */
+    ClearSuffix,     /**< Clear both -.... and +.... information during bump */
+    ClearPreRelease, /**< Clear the -.... information during bump */
+    ClearBuildMeta   /**< Clear the +.... information during bump */
+};
+
+/**
+ * @enum ReleaseType
+ * Enumerates different release types for semantic versioning.
+ */
+enum class ReleaseType {
+    Stable,          /**< Stable release with no extra pre-release or build metadata */
+    Alpha,           /**< Alpha pre-release: x.y.z-alpha.... */
+    Beta,            /**< Beta pre-release: x.y.z-beta.... */
+    AlphaBeta,       /**< Alpha-Beta pre-release: x.y.z-alpha.beta.... */
+    PreRelease,      /**< Custom pre-release: x.y.z-prerelease.... */
+    ReleaseCandidate /**< Release Candidate: x.y.z-rc.... */
+};
+
+/**
+ * @enum CompareType
+ * Enumerates different ways to compare semantic versions.
+ */
+enum class CompareType {
+    Exact, /**< Exact match including suffix */
+    Major, /**< Compare only major version */
+    Minor, /**< Compare only minor version */
+    Patch  /**< Compare only patch version */
+};
+
+/**
+ * @class SemVer
+ * Represents a Semantic Version.
+ */
+class SemVer {
+    char const m_mmp_sep;
```

Comment:
```suggestion
    char const m_number_separator;
```

This should also most likely be inside one of the access modifiers.